src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -0,0 +1,662 @@
+
+DROP TABLE batch_sn_info;
+	)");
+
+	transaction.commit();
+
+  create_schema();
+
+	MINFO("Database reset complete");
+}
+
+void BlockchainSQLite::load_database(std::optional<fs::path> file) {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
+  this->height = 0;
+  if (db)
+    throw std::runtime_error("Reloading database not supported");
+
+  std::string fileString;
+  if (file.has_value())
+  {
+    fileString = file->string();
+    MINFO("Loading sqliteDB from file " << fileString);
+  }
+  else
+  {
+    fileString = ":memory:";
+    MINFO("Loading memory-backed sqliteDB");
+  }
+  db = std::make_unique<SQLite::Database>(
+      SQLite::Database{
+      fileString, 
+      SQLite::OPEN_READWRITE | SQLite::OPEN_CREATE | SQLite::OPEN_FULLMUTEX,
+      SQLite_busy_timeout.count()
+      });
+
+  if (!db->tableExists("batch_sn_payments") || !db->tableExists("batch_sn_info") || !db->tableExists("block_payments")) {
+    create_schema();
+  }
+
+  SQLite::Statement st{*db, "SELECT height FROM batch_sn_info"};
+  while (st.executeStep()) {
+    this->height = st.getColumn(0).getInt64();
+  }
+  filename = fileString;
+}
+
+bool BlockchainSQLite::update_height(uint64_t new_height) {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__ << " Called with new height: " << new_height);
+  try {
+    SQLite::Transaction transaction{*db};
+    this->height = new_height;
+    SQLite::Statement update_height{*db,
+      "UPDATE batch_sn_info SET height = ?"};
+    exec_query(update_height, static_cast<int64_t>(height));
+
+    // Commit transaction
+    transaction.commit();
+  }
+  catch (std::exception& e)
+  {
+    MFATAL("exception: " << e.what());
+    return false;
+  }
+  return true;
+}
+
+bool BlockchainSQLite::increment_height() {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__ << " Called with height: " << this->height + 1);
+  return update_height(this->height + 1);
+}
+
+bool BlockchainSQLite::decrement_height() {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__ << " Called with height: " << this->height - 1);
+  return update_height(this->height - 1);
+}
+
+uint64_t BlockchainSQLite::batching_count() {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
+  SQLite::Statement st{*db, "SELECT count(*) FROM batch_sn_payments"};
+  uint64_t count = 0;
+  while (st.executeStep()) {
+    count = st.getColumn(0).getInt64();
+  }
+  return count;
+}
+
+std::optional<uint64_t> BlockchainSQLite::retrieve_amount_by_address(const std::string& address) {
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
+  SQLite::Statement st{*db, "SELECT amount FROM batch_sn_payments WHERE address = ?"};
+  st.bind(1, address);
+  std::optional<uint64_t> amount = std::nullopt;
+  while (st.executeStep()) {
+    assert(!amount);
+    amount.emplace(st.getColumn(0).getInt64());
+  }
+  return amount;
+}
+
+bool BlockchainSQLite::add_sn_payments(cryptonote::network_type nettype, std::vector<cryptonote::batch_sn_payment>& payments, uint64_t block_height)
```

Comment:
Could we maybe specify this network_type in the constructor (and store it as a member) so that we don't have to pass it in these various functions?  You can't ever change it, so this is always passing in the same value.